### ISSUE TITLE
Avoid making ISSHConnectionCreator public just yet

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6541]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6541|
|Type|task|
|Created|2013-06-01 18:15:17Z|
|Branch|https://github.com/twisted/twisted/tree/private-conn-creator-interface-6541|

`ISSHConnectionCreator` was introduced along with `twisted.conch.endpoints` as a related API for creating the two endpoints defined by that module.

In the course of [#6532](https://github.com/twisted/twisted/issues/6532) it has become apparent that the interface may not be sufficiently expressive to allow cancellation to be implemented.

To avoid needing to introduce a new interface or find some way to make a backwards compatible change to `ISSHConnectionCreator`, it should be made private before it gets included in a release - so that we are not committed to supporting it *and* so that [[[#6532](https://github.com/twisted/twisted/issues/6532)](https://github.com/twisted/twisted/issues/6532)](https://github.com/twisted/twisted/issues/6532) does not block an upcoming release.

The goal should be to make it public again soon after #6532 is resolved (or as part of #6532 itself).

<details><summary>Searchable metadata</summary>

```
trac-id__6541 6541
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_private_conn_creator_interface_6541 branches/private-conn-creator-interface-6541
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1370110517000000 1370110517000000
changetime__1370266460000000 1370266460000000
version__None None
owner__exarkun exarkun
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [38608]) Branching to 'private-conn-creator-interface-6541'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Build results - http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/private-conn-creator-interface-6541

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_4"></a>|@jonathanj set owner to @exarkun|
|-|-|

Looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [38628]) Merge private-conn-creator-interface-6541

Author: exarkun
Reviewer: jonathanj
Fixes: [#6541](https://github.com/twisted/twisted/issues/6541)

Change `ISSHConnectionCreator` to `_ISSHConnectionCreator` so as not to make it a
part of the public interface of `twisted.conch.endpoints`, pending some changes we
probably want to make to it (at least for cancellation - see [#6532](https://github.com/twisted/twisted/issues/6532)).
